README.md

```diff
@@ -4,47 +4,97 @@
 uftrace
 =======
 
-The uftrace tool is to trace and analyze execution of a program
-written in C/C++.  It was heavily inspired by the ftrace framework
-of the Linux kernel (especially function graph tracer) and supports
-userspace programs.  It supports various kind of commands and filters
-to help analysis of the program execution and performance.
+uftrace is a function call graph tracer for C, C++, Rust and Python programs.
+Using dynamic tracing, it can include nested library calls and
+even seamless function graph tracing of used kernel functions.
+
+- Shows colored nested function call graphs (instead of listings)!
+- Displays arguments symbolically using libc function prototypes and DWARF
+  debug information values.
+- Filters like minimal function call duration are available
+- Traces trace kernel events e.g. scheduling events (which affecting the
+  execution timing of the program) and records nanosecond-exact timestamps.
+
+uftrace traces, records and analyzes the execution of programs written in
+C, C++ and Rust. It was heavily inspired by the ftrace framework of the
+Linux kernel (especially the function graph tracer) and supports userspace
+programs, and with elevated privileges, also kernel function call graphs.
+
+The name uftrace stems from the combination of user and ftrace.
+
+It supports many commands and filters such as filtering by function call
+duration for analysis of program execution and performance.
 
 ![uftrace-live-demo](doc/uftrace-live-demo.gif)
 
  * Homepage: https://github.com/namhyung/uftrace
  * Tutorial: https://github.com/namhyung/uftrace/wiki/Tutorial
  * Chat: https://gitter.im/uftrace/uftrace
  * Mailing list: [uftrace@googlegroups.com](https://groups.google.com/forum/#!forum/uftrace)
+ * Lightning talk: https://youtu.be/LNav5qvyK7I
 
 
 Features
 ========
 
-It traces each function in the executable and shows time duration.  It
-can also trace external library calls - but usually entry and exit are
-supported.  Optionally it's possible to trace other (nested) external
-library calls and/or internal function calls in the library call.
-
-It can show detailed execution flow at function level, and report which
-function has the highest overhead.  And it also shows various information
-related the execution environment.
+uftrace traces each function in the executable and shows time durations.
+
+Usually, for this to be possible, the program needs to be compiled with
+`-pg` or `-fpatchable-function-entry=2`, but with fully dynamic tracing
```

Comment:
The number of NOPs required depends on architecture.

`-fpatchable-function-entry=2` is only for AArch64. For x86_64, it must be `-fpatchable-function-entry=5`.

Reply:
I'll update it to use "full dynamic tracing" to be consistent but grammar-wise, it looks a bit strange:

"full" is mostly used to refer to a container or all of the something which has some boundary, like:
full moon, strength, full speed, full support, full report.

It is used to refer something which can be "partial", "half", or "minimal", like half moon, half speed, partial report, or minimal support.

See the Cambridge dictionary, for example:
https://dictionary.cambridge.org/dictionary/english/full
https://www.merriam-webster.com/dictionary/full

I'm not saying that "fully" is correct, because I know no definition of "full dynamic tracing" nor of "fully dynamic tracing". I don't think either one is good, I think "fully" just sounds less wrong. Omitting "full" and "fully" would be best I think, but that is a global change.

I know that "dynamic" in regarding to libraries and symbols refers to shared libraries which are dynamically linked into programs, but I do not actually know if "dynamic" tracing:

I have two interpretations:
- Tracing of shared (dynamically loaded) libraries.
- Tracing which can be dynamically configured (enabled, disabled, refined, changed, reconfigured)

I assume the latter. But what is the difference beween "dynamic tracing" (like normal? dynamic tracing" and "full dynamic tracing"?